src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttribute.cs

```diff
@@ -87,5 +82,53 @@ public CosmosDBTriggerAttribute(string databaseName, string collectionName)
         /// collection.
         /// </summary>
         public int LeasesCollectionThroughput { get; set; }
+
+        /// <summary>
+        /// Optional.
+        /// Defines a prefix to be used within a Leases collection for this Trigger. Useful when sharing the same Lease collection among multiple Triggers
+        /// </summary>
+        public string LeaseCollectionPrefix { get; set; }
+
+        /// <summary>
+        /// Optional.
+        /// Customizes the amount of milliseconds between lease checkpoints
+        /// </summary>
+        public int CheckpointFrequency { get; set; }
+
+        /// <summary>
+        /// Optional.
+        /// Customizes the amount of documents between lease checkpoints
+        /// </summary>
+        public int CheckpointFrequencyProcessedDocuments { get; set; }
```

Comment:
In the Change Feed Processor logic, if the user sets both, the Checkpoint will happen if either condition is met (either by time or by amount). By default, Checkpoint happens after every execution, but these settings allow the user to customize it for a time interval or a processed documents amount (instead of on every execution).